airbyte-oauth/src/main/java/io/airbyte/oauth/flows/SurveymonkeyOAuthFlow.java

```diff
@@ -36,14 +45,26 @@ public SurveymonkeyOAuthFlow(final ConfigRepository configRepository, final Http
     super(configRepository, httpClient, stateSupplier);
   }
 
+  protected String getAuthorizeUrl(final JsonNode inputOAuthConfiguration) {
+    final String origin = getConfigValueUnsafe(inputOAuthConfiguration, "origin");
+    if (origin.equals(EUROPE)) {
+      return API_ACCESS_URL_EU + AUTHORIZE_URL;
+    }
+    if (origin.equals(CANADA)) {
+      return API_ACCESS_URL_CA + AUTHORIZE_URL;
+    }
+    return API_ACCESS_URL_USA + AUTHORIZE_URL;
```

Comment:
this should fail if "origin" != "USA" (eg if we add an enum value and forget to update this block)

Reply:
the reason we added origin is a different urls for authorization, USA is default, but if someone wants to add new origin he should know about it, because it's origin purpose. I can add comment in spec.json about it.